### PR TITLE
Add module listing page with required tooling

### PR DESCRIPTION
**Pull request by chetanmeh**
This PR adds tooling to generate a `modules.md` page which lists all OpenWhisk related repositories along with there Build status and links to navigate to those repositories.

See current rendition in my fork [here](https://github.com/chetanmeh/incubator-openwhisk/blob/list-ow-projects/docs/dev/modules.md)

This is inspired from [Sling Module Listing](https://github.com/apache/sling-aggregator/blob/master/docs/modules.md) and provides a consolidated listing of all modules in a single place

## Description

While working on #3823 I felt a need to have an overview of build status of all OpenWhisk repositories in a single place. As part of this PR 2 dev tasks are being added

* `:tools:dev:listRepos` - Uses Github search api to list all OpenWhisk related repositories and dumps them to a file
* `:tools:dev:renderModuleDetails` - Use previous dump to render a `modules.md` page under `docs/dev` which can be checked in

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [x] General tooling
- [x] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [x] My changes require further changes to the documentation.
- [x] I updated the documentation where necessary.

### CODE REVIEW

File: docs/dev/modules.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+|---	|---	|
+| [incubator-openwhisk-release](https://github.com/apache/incubator-openwhisk-release) | Apache openwhisk release |
+| [incubator-openwhisk-utilities](https://github.com/apache/incubator-openwhisk-utilities) | Shared utilities used across Apache OpenWhisk project repositories. |
+
+
+## Others
+
+| Module | Description |
+|---	|---	|
+| [incubator-openwhisk-external-resources](https://github.com/apache/incubator-openwhisk-external-resources) | ? Curated list of awesome OpenWhisk things ? |
```
Comment (dubee) [inline docs/dev/modules.md]: Don't think we need the question marks here.

Comment (chetanmeh) [reply docs/dev/modules.md]: Its taking description of repo verbatim. And this repo has description like 
✨ Curated list of awesome OpenWhisk things ✨

Would try to improve the handling here

Comment (chetanmeh) [reply docs/dev/modules.md]: This is fixed now

**Comment by chetanmeh** [comment]
Need a PG for this

**Comment by chetanmeh** [comment]
Can someone trigger a PG for this?

**Comment by chetanmeh** [comment]
Can someone trigger a PG for this? (take 2)

**Comment by rabbah** [comment]
@dubee any concerns with this otherwise i think we should merge it.

**Comment by cbickel** [comment]
PG1#3121 (with rev 33538c0e40b513ec8b19563f4ff20b37022f0e86) is 🔵

**Pull request merged**